### DISCUSSION TITLE
Performance test with jmeter

### DISCUSSION THREAD

**Discussion by jkc20** [zauberzeug/nicegui#3117] [Q&A]
### Question

In my app I use ui.download which produce a static link to the file.  The analysis of  the html request shows a link with the following structure:
/_nicegui/auto/static/some-kind-of-uuid/filename
The question is, how one can programaticaly get the "some-kind-of-uuid" part and build the download path for the performance test automatically ?

**Comment by falkoschindler** [comment]
Hi @jkc20,

I have difficulties to understand what you're trying to achieve. What do you mean with

> programmatically get the "some-kind-of-uuid" part and build the download path for the performance test

Do you need the server to somehow detect the HTTP requests and act depending on the UUID? Then a middleware might be what you're looking for.

**Comment by jkc20** [comment]
Hi Falko,
tx for the fast response. I've build a converter as webservice to convert csv data to a pdf file.  My customer want to use it in a cloud environment, where at least 100 of his customers use the service simultanious. Therefor my customer want to do some load testing with jmeter. He analyses the traffic with fiddler and figured out, that the download path changes on every access:
/_nicegu/auto/32-digit-hex-number/filename. The 32-digit-hex-number changes on every request.  Do you mean fastapi by middleware ?

**Reply by falkoschindler** [reply]
Yes, FastAPI has the concept of middleware which you can implement to intercept requests.

To clarify things, I suggest to provide some [example code](https://en.wikipedia.org/wiki/Minimal_reproducible_example): How is `ui.download` used? What is downloaded? How are the files or the data served? What exactly do you want to happen?

**Comment by jkc20** [comment]
I'm totally new to load test, just played a little bit with locust. If I got it right, one analyses the webtraffic with tools like fiddler and build upon of that a testscript where the variable parts of the traffic will programmatically adapted and so, one can sent 10 or 100 request a second. For this approach they need to build the correct download path for every request.

**Comment by jkc20** [comment]
Thanks for now, I guess the generate_pdf from the example folder shout work. But I need to investigate in Jmeter to produce the question point of my customer. I guss that will take a while.

**Comment by jkc20** [comment]
My customer give some more information - the get request sniffed with fiddler (no option for me, Windows and not open source):
GET [https://SERVERNAME/_nicegui/auto/static/a900dc1656e479d92dcaf815b16076b1/](https://SERVERNAME/_nicegui/auto/static/a900dc1656e479d92dcaf815b16076b1/FILENAME and the next request
GET https://SERVERNAME/_nicegui/auto/static/3d0ec2520ea177119f4721e0d6509a7c/FILENAME 
I guess it would be better to provide the customer my on loadtest solution with locust and its selenium plugin 
close it for now
